### PR TITLE
Fixed: NULL remoteHost in IOCTL_SO_INETATON

### PR DESCRIPTION
**Pull request by sepalani**
Fix a crash when a domain name doesn't exist. (Ex: Monster Hunter Tri)

### CODE REVIEW

**Comment by Tilka** [comment]
Not pretty, but that's not your fault... lgtm

File: Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_net.cpp [hunk 1]
```diff
@@ -921,11 +921,21 @@ IPCCommandResult CWII_IPC_HLE_Device_net_ip_top::IOCtl(u32 _CommandAddress)
...
+			INFO_LOG(WII_IPC_NET, "IOCTL_SO_INETATON = %d "
+				"%s, BufferIn: (%08x, %i), BufferOut: (%08x, %i), IP Found: NULL", -1,
+				hostname.c_str(), BufferIn, BufferInSize, BufferOut, BufferOutSize);
+			ReturnValue = 0;
+		}
+		else
+		{
+			Memory::Write_U32(Common::swap32(*(u32*)remoteHost->h_addr_list[0]), BufferOut);
+			INFO_LOG(WII_IPC_NET, "IOCTL_SO_INETATON = %d "
+				"%s, BufferIn: (%08x, %i), BufferOut: (%08x, %i), IP Found: %08X", remoteHost->h_addr_list[0] == nullptr ? -1 : 0,
```
Comment (degasus) [inline Source/Core/Core/IPC_HLE/WII_IPC_HLE_Device_net.cpp]: We can try to clean it up a bit: I guess IOCTL_SO_INETATON = %d shall print the return value? At least I'm for doing so.
And if you're able to fit the last value, we can move this INFO_LOG out of the if/else.

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

The last value can be set to INADDR_NONE (i.e. -1) when remoteHost is NULL. I'm not a huge fan of it because it's a valid broadcast address, but it's only for INFO_LOG so it should be fine.

**Comment by JMC47** [comment]
@dolphin-emu-bot rebuild

**Comment by Tilka** [comment]
Oh, and both the old and the current code dereference `remoteHost->h_addr_list[0]` and then check for null.

**Comment by BhaaLseN** [comment]
If you're doing a cleanup anyways, why not make it an ```if nullptr```/```else``` with two distinct logs?

**Comment by sepalani** [comment]
There is no normal way for ```remoteHost->h_addr_list``` to be NULL in most Linux implementations:
```#define h_addr h_addr_list[0] /* for backward compatibility */```
I don't know if that's still the case with outlandish implementation so I checked it anyway.

Otherwise, making two distinct logs can be a good idea and it would be slightly faster than those ternaries I guess. What would you advise me to do?

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

**Comment by BhaaLseN** [comment]
Its unlikely to be noticably any faster or slower; my argument is for readibilty.

**Comment by sepalani** [comment]
Fine.

Looks like my first commit but with more checks which allowed to get rid of those ternaries and unnecessary string formats by the same way.

However I don't know how this kind of changes can impact graphical rendering...

**Comment by Tilka** [comment]
Ignore dx-win-nv, it's currently broken.

**Pull request merged**